### PR TITLE
fix: Fix certain sibling tasks not connected to parent

### PR DESCRIPTION
**Pull request by simster7**
Fixes: https://github.com/argoproj/argo-workflows/issues/6172

Signed-off-by: Simon Behar <simbeh7@gmail.com>

Checklist:

* [X] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

Tips:

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it does not need to pass
* Sign-off your commits to pass the DCO check: `git commit --signoff`.
* Run `make pre-commit -B` to fix codegen or lint problems. 
* Say how how you tested your changes. If you changed the UI, attach screenshots.

### CODE REVIEW

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -2266,9 +2266,9 @@ func (woc *wfOperationCtx) executeContainer(ctx context.Context, nodeName string
 func (woc *wfOperationCtx) getOutboundNodes(nodeID string) []string {
 	node := woc.wf.Status.Nodes[nodeID]
 	switch node.Type {
-	case wfv1.NodeTypeSkipped, wfv1.NodeTypeSuspend:
+	case wfv1.NodeTypeSkipped, wfv1.NodeTypeSuspend, wfv1.NodeTypePod:
```
Comment (simster7) [inline workflow/controller/operator.go:2269]: @alexec You changed this in #5099 (https://github.com/argoproj/argo-workflows/pull/5099/files#diff-f321d4af83fcf8311dc80c0d50c59ac4c240f761206e7bb652709870eb9feb43R2260)

I believe that this is the correct code, but I may be missing something about how container sets are linked together. Can you please review?

Comment (alexec) [reply workflow/controller/operator.go:2269]: I can't remember.

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
Please lint!

**Comment by simster7** [comment]
I just ran some tests, this needs a tad more work. Please don't merge.

File: workflow/controller/operator.go (hunk 1 above)
Comment (simster7) [reply workflow/controller/operator.go:2269]: I was wrong, this fix will introduce a small break to container sets. I'll find a fix that works for both.

Comment (simster7) [reply workflow/controller/operator.go]: > I'll find a fix that works for both.

Fixed

**Comment by simster7** [comment]
> this needs a tad more work

Done

**Review by alexec — APPROVED** [review]

**Pull request merged**